File: xpeapp_android/app/src/main/java/com/xpeho/xpeapp/data/model/Newsletter.kt
package com.xpeho.xpeapp.data.model

import com.google.firebase.firestore.ServerTimestamp
import java.time.LocalDate

data class Newsletter(
    val id: String,
    val summary: String,
    @ServerTimestamp
    val date: LocalDate,
    @ServerTimestamp
    val publicationDate: LocalDate,
    val pdfUrl: String,
)

File: xpeapp_ios/XpeApp/XpeApp/Features/Newsletter/model/Newsletter.swift
//
//  Newsletter.swift
//  XpeApp
//
//  Created by Ryan Debouvries on 23/08/2024.
//

import Foundation

struct Newsletter: Codable {
    var id: String?
    let date: Date
    let pdfUrl: String
    let publicationDate: Date
    let summary: String
    
    init(id: String? = nil, date: Date, pdfUrl: String, publicationDate: Date, summary: String) {
        self.id = id
        self.date = date
        self.pdfUrl = pdfUrl
        self.publicationDate = publicationDate
        self.summary = summary
    }
}
